rpc-rs/src/provider_main.rs

```diff
@@ -288,8 +275,18 @@ where
     }
 }
 
-#[cfg(not(feature = "otel"))]
-fn configure_tracing(_: String, structured_logging_enabled: bool) {
+/// Configure tracing for logs or otel, depending on "otel" feature flag
+pub fn configure_tracing(provider_name: String, structured_logging_enabled: bool) {
```

Comment:
You didn't pick this name, but I'm wondering if there's a better one before we publish a provider that uses this name.

"Tracing" is overloaded in this context: there's OTEL tracing, and there's the tracing crate. We use the tracing crate to register a global subscriber and emit logs irrespective of whether OTEL tracing is enabled, so from that perspective it makes sense to "configure [using the] tracing [crate]." But if the otel feature flag is disabled, we're not "configuring [OTEL] tracing."

Thoughts?

Reply:
I'm not sure I see the conflict. I read it as configure the kind of tracing,  not configure tracing on or off.  If otel is disabled, the tracing enabled by "configure_tracing" will be the regular kind with tracing crate and outputs to stdout. if otel is enabled, the tracing will be emitted to otel.  I'm open to other names if you have a suggestion for something clearer.